### PR TITLE
Use Lookup widget (Search) instead of listview for all windows

### PR DESCRIPTION
**Pull request by riadmen**
https://github.com/metasfresh/metasfresh/issues/18285

### CODE REVIEW

File: backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/system/70-de.metas.fresh/5726940_sys_gh18285_Use_Lookup_Instead_Listview.sql [hunk 1]
```diff
@@ -0,0 +1,16 @@
+select backup_table('ad_column'); -- backup.ad_column_bkp_20240620_114638_818
+select backup_table('ad_field'); -- backup.ad_field_bkp_20240620_114658_431
+
+UPDATE AD_Field f
+SET AD_Reference_ID = 30
+WHERE f.AD_Reference_ID IN (18, 19)
+  AND f.AD_Column_ID NOT IN (SELECT AD_Column_ID
+                             FROM AD_Column
+                             WHERE AD_Reference_ID = 30)
```
Comment (teosarca) [inline backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/system/70-de.metas.fresh/5726940_sys_gh18285_Use_Lookup_Instead_Listview.sql:9]: so this UPDATE will update all fields
and will set reference=Search
if 
* the field reference is Table,TableDir (i.e. not Search)
* and the column reference is Search

Is that matching the logic provided in requirements? i.e.
image

Comment (riadmen) [reply backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/system/70-de.metas.fresh/5726940_sys_gh18285_Use_Lookup_Instead_Listview.sql:9]: yes

Comment (teosarca) [reply backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/system/70-de.metas.fresh/5726940_sys_gh18285_Use_Lookup_Instead_Listview.sql:9]: not so sure... am i missing something?

image

Comment (riadmen) [reply backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/system/70-de.metas.fresh/5726940_sys_gh18285_Use_Lookup_Instead_Listview.sql:9]: I understand it like this: `Exclude fields (leave them as list) where fields != search and column=search`:

Comment (teosarca) [reply backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/system/70-de.metas.fresh/5726940_sys_gh18285_Use_Lookup_Instead_Listview.sql:9]: that's right.

now, can u pls check the SQL bellow and pls tell me what's that doing compared to yours:
```sql
UPDATE AD_Field f SET AD_Reference_ID = 30
FROM AD_Column c
WHERE f.AD_Column_ID=c.AD_Column_ID
    AND f.AD_Reference_ID IS NOT NULL
    AND f.AD_Reference_ID=c.AD_Reference_ID
;
```

Comment (teosarca) [reply backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/system/70-de.metas.fresh/5726940_sys_gh18285_Use_Lookup_Instead_Listview.sql:9]: also, before updating you could check and document which are the affected cases.... shall not be so many

Comment (riadmen) [reply backend/de.metas.fresh/de.metas.fresh.base/src/main/sql/postgresql/system/70-de.metas.fresh/5726940_sys_gh18285_Use_Lookup_Instead_Listview.sql]: the list of fields that wil be updated are documented in the dev docu

**Pull request merged**